cli/config/configfile/file.go

```diff
@@ -263,10 +290,61 @@ func decodeAuth(authStr string) (string, string, error) {
 // GetCredentialsStore returns a new credentials store from the settings in the
 // configuration file
 func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) credentials.Store {
+	store := credentials.NewFileStore(configFile)
+
 	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
-		return newNativeStore(configFile, helper)
+		store = newNativeStore(configFile, helper)
+	}
+
+	envConfig := os.Getenv(dockerEnvConfig)
+	if envConfig == "" {
+		return store
+	}
+
+	// use DOCKER_AUTH_CONFIG if set
+	// if a parse error occurs it falls back to the native or file store
+	envStore, err := memorystore.New(
+		withEnvConfig(envConfig),
```

Comment:
The `withEnvConfig` seems to be a bit weird wrapper around `memorystore.WithAuthConfig` that is doing two separate things (parsing the env string AND then acting as an functional option for the memorystore).

It would be much clearer if this was just an explicit `parseEnvConfig` function that takes an string and returns the parsed object which is then passed to `memorystore.WithEnvConfig`.

Reply:
Something like:

```suggestion
	ec, err := parseEnvConfig(envConfig)
	if err != nil {
		return fmt.Errorf("failed to parse DOCKER_AUTH_CONFIG: %v", err)
	}
	
	// use DOCKER_AUTH_CONFIG if set
	// if a parse error occurs it falls back to the native or file store
	envStore, err := memorystore.New(
		memorystore.WithEnvConfig(ec),
```